mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivateContents.java

```diff
@@ -0,0 +1,547 @@
+            super(p, attr);
+        }
+
+        @XmlElement(name = "directory")
+        @JsonProperty("directories")
+        @JsonInclude(content = JsonInclude.Include.NON_EMPTY)
+        public List<Directory> getDirs() {
+            return dirs;
+        }
+
+        @XmlElement(name = "file")
+        @JsonProperty("files")
+        @JsonInclude(content = JsonInclude.Include.NON_EMPTY)
+        public List<File> getFiles() {
+            return files;
+        }
+
+    }
+
+    private static class File extends DirectoryEntry {
+
+        private String md5;
+
+        private long size;
+
+        File() {
+            super();
+        }
+
+        File(MCRPath p, MCRFileAttributes attr) {
+            super(p, attr);
+            this.md5 = attr.md5sum();
+            this.size = attr.size();
+        }
+
+        @XmlAttribute
+        @JsonProperty(index = 3)
+        public String getMd5() {
+            return md5;
+        }
+
+        @XmlAttribute
+        @JsonProperty(index = 1)
+        public long getSize() {
+            return size;
+        }
+    }
+
+    @JsonInclude(content = JsonInclude.Include.NON_NULL)
+    private abstract static class DirectoryEntry implements Comparable<DirectoryEntry> {
+        private String name;
+
+        private Date modified;
+
+        DirectoryEntry(MCRPath p, MCRFileAttributes attr) {
+            this.name = Optional.ofNullable(p.getFileName())
+                .map(java.nio.file.Path::toString)
+                .orElse(null);
+            this.modified = Date.from(attr.lastModifiedTime().toInstant());
+        }
+
+        DirectoryEntry() {
+        }
+
+        @XmlAttribute
+        @JsonProperty(index = 0)
+        @JsonInclude(content = JsonInclude.Include.NON_EMPTY)
+        String getName() {
+            return name;
+        }
+
+        @XmlAttribute
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = MCRRestUtils.JSON_DATE_FORMAT)
+        @JsonProperty(index = 2)
+        public Date getModified() {
+            return modified;
+        }
+
+        @Override
+        public int compareTo(DirectoryEntry o) {
+            return Ordering
+                .<DirectoryEntry> from((de1, de2) -> {
+                    if (de1 instanceof Directory && !(de2 instanceof Directory)) {
+                        return -1;
+                    }
+                    if (de1.getClass().equals(de2.getClass())) {
+                        return 0;
+                    }
+                    return 1;
+                })
+                .compound((de1, de2) -> de1.getName().compareTo(de2.getName()))
+                .compare(this, o);
+        }
+    }
+
+    private static class MaxBytesOutputStream extends CountingOutputStream {
+
+        private final long maxSize;
+
+        MaxBytesOutputStream(OutputStream out) {
+            super(out);
+            maxSize = MCRConfiguration2.getOrThrow("MCR.FileUpload.MaxSize", Long::parseLong);
```

Comment:
A constant would be nice (even if there are only 2 places).

Reply:
constant could not change during runtime.  Externalized methods for upload configuration instead